python/src/scipp/utils/dimensions.py

```diff
@@ -0,0 +1,21 @@
+# SPDX-License-Identifier: BSD-3-Clause
+# Copyright (c) 2021 Scipp contributors (https://github.com/scipp)
+# @author Jan-Lukas Wynen
+
+
+def find_bin_edges(var, ds):
+    """
+    Returns a list of bin-edge dimension labels assuming that
+    var is a coord or attr of ds.
+    """
+    bin_edges = []
+    for idx, dim in enumerate(var.dims):
+        length = var.shape[idx]
+        if not ds.dims:
+            # Have a scalar slice.
+            # Cannot match dims, just assume length 2 attributes are bin-edge
```

Comment:
I don't follow the logic for his. Why length 2?

Reply:
It is not possible to determine if the coord / attr is bin-edge if its dim does not match any in the data. We chose to identify `length == 2` as bin-edge because that case comes about when slicing to scalar.

Reply:
Ok this is slightly more robust than I thought once we work on the assumption that var must be part of the metadata.